### PR TITLE
Kuadrant Release Process

### PR DESCRIPTION
**Pull request by didierofrivia**
Partially closes https://github.com/Kuadrant/architecture/issues/59

Technical aspects of the process: https://github.com/Kuadrant/architecture/pull/41

### CODE REVIEW

File: rfcs/0000-kuadrant-release-process.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+are ready for testing. This notification can be done through comments on GitHub issues or via other communication channels
+like Slack.
+5. Testing: The QA team tests the release candidate, checking for any bugs or issues. They can use GitHub's issue tracking
+system to report any problems they find.
+6. Iterate: Based on the feedback from the QA team, the development team makes any necessary adjustments and repeats the
+process until the release candidate is deemed ready for production.
+7. Publish Release: Once the release candidate is ready, the engineering team publishes the release. This means that the
+Github release draft is published, and the artifacts are uploaded to the corresponding registries.
+
+## Cadence
```
Comment (Boomatang) [inline rfcs/0000-kuadrant-release-process.md:92]: This question might not belong here but looking through the links I was not able to find the answer or the correct place for the question. 

If we are releasing the Kuadrant Operator every two weeks, what is our support policy? How many version are we going to back port security and bug fixes to?

Comment (didierofrivia) [reply rfcs/0000-kuadrant-release-process.md:92]: It's a valid question, and important one to have it registered in the RFC. I'd say that we should include not only engineering and QA, but what would be the expectation from Product as well.

File: rfcs/0000-kuadrant-release-process.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+Once the project is stable enough, and it's adoption increases, the community will be expecting a certain degree of
+commitment from the maintainers, and that includes a regular release cadence. The frequency of the releases of the
+different components could vary depending on the particular component needs. However, the **Kuadrant Operator**
+it's been discussed in the past that it should be released every 2 weeks, including the latest released version
```
[the remark this answers is not in the document]
Comment (didierofrivia) [reply rfcs/0000-kuadrant-release-process.md:97]: Will reword it was you mention

File: rfcs/0000-kuadrant-release-process.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+- Feature Name: `kuadrant-release-process`
+- Start Date: 2024-01-11
+- RFC PR: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/pull/0000)
+- Issue tracking: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Kuadrant is a set of components that their artifacts are built and delivered independently. This RFC aims to define every
+aspect of the event of releasing a new version of the whole, in terms of versioning, cadence, communication, channels,
```
[the remark this answers is not in the document]
Comment (didierofrivia) [reply rfcs/0000-kuadrant-release-process.md:12]: I reckon that Paul's one will address the actual definition of cadence. Say, for some components, for the whole, etc. Kind of agreeing on the frequency. 
And in this one, the importance of it. Does it make sense?

File: rfcs/0000-kuadrant-release-process.md [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+* MAJOR version when you make incompatible API changes
+* MINOR version when you add functionality in a backward compatible manner
+* PATCH version when you make backward compatible bug fixes
+
+Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.
+
+A more detailed explanation of the versioning scheme can be found in the [Semantic Versioning](https://semver.org/) website.
+
+By releasing a new version of Kuadrant, we mean releasing a new version of every component in the suite, with their
```
[the remark this answers is not in the document]
Comment (didierofrivia) [reply rfcs/0000-kuadrant-release-process.md:58]: Will reword it stating that some components might still be the same as the previous release.

File: rfcs/0000-kuadrant-release-process.md [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+
+- Feature Name: `kuadrant-release-process`
+- Start Date: 2024-01-11
+- RFC PR: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/pull/0000)
+- Issue tracking: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Kuadrant is a set of components that their artifacts are built and delivered independently. This RFC aims to define every
```
[the remark this answers is not in the document]
Comment (didierofrivia) [reply rfcs/0000-kuadrant-release-process.md:11]: I think that the difference is Kuadrant operator (the orchestrator of the suite) and Kuadrant components (Authorino, Limitador, MGC, etc). I'm saying this because I thought that QA mainly checks the versions included in the latest released version of Kuadrant Operator. I could re write it as per each component... which kind of will remove the responsibility of deciding the cadence and leave it to Pauls RFC (?)

File: rfcs/0000-kuadrant-release-process.md [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+* MAJOR version when you make incompatible API changes
+* MINOR version when you add functionality in a backward compatible manner
+* PATCH version when you make backward compatible bug fixes
+
+Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.
+
+A more detailed explanation of the versioning scheme can be found in the [Semantic Versioning](https://semver.org/) website.
+
+By releasing a new version of Kuadrant, we mean releasing a new version of every component in the suite, with their
+corresponding semantic versioning, and being the version of the **Kuadrant Operator** the one that defines the version of
```
[the remark this answers is not in the document]
Comment (didierofrivia) [reply rfcs/0000-kuadrant-release-process.md:59]: Sounds like a good candidate for the community call indeed. It's something I kind of been suggesting, or at least a given xD

File: rfcs/0000-kuadrant-release-process.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (didierofrivia) [reply rfcs/0000-kuadrant-release-process.md:11]: On another note, it mentions that the artifacts from each component are build (and delivered) independently, but might not be subject to a new event of releasing.

File: rfcs/0000-kuadrant-release-process.md (hunk 1 above)
Comment (didierofrivia) [reply rfcs/0000-kuadrant-release-process.md:92]: I will add it to the "Unresolved questions" section

File: rfcs/0000-kuadrant-release-process.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (guicassolato) [reply rfcs/0000-kuadrant-release-process.md:11]: > actual release process should be the same for every component

I see this differently.

IMO, it does not _have to_ be, neither it _can_ always be the same process for every component. Rather (as originally phrased), the components are independent from the whole and can be released independently, following their own particular release processes, respecting each one its own specificity. It's not top-down; it's bottom-up.

For some component, it may make sense, for example, doing release branches, while for others it's not needed and just a tag will do. Some components may involve publishing artifacts to a particular repo, while others do not. Because of these and so many other differences, each component is a project and has its own history. Thinking the opposite risks creating coupling between the components and generates scalability issues to the organisation.

Of course, we don't want multiple ways to do the exact same thing. To reduce overhead, if 2 components share a common step within their corresponding release processes, we should try performing that step in the same way – by resusing tools, automation, docs, etc across components. (Although, I think this would be something that belongs to #41 and not to this PR.)

Correct if I'm wrong, but I understand this RFC is for how to lock a version of what we call "Kuadrant", which is not the sum of its parts, but has an identity of its own. Therefore, it's not to force components all to stick to a same process but to highlight precisely this difference between the sum of the parts and the whole. Maybe it's a description of the steps to follow _after_ all components have been released, and perhaps some general guidelines that all components need to comply with, otherwise releasing Kuadrant would be impossible.

If there's a components that confuses itself with the whole, then an explanation of that and some words about how releasing that one component wraps up a release of Kuadrant itself maybe (if that's the case.)

[the remark this answers is not in the document]
Comment (pehala) [reply rfcs/0000-kuadrant-release-process.md:11]: I would say this RFC should be high-level and only mention the release process, not the actual steps, e.g. how we build those artefacts should not be part of this (That is part of the Kuadrant CI/CD RFC and that indeed differs). I do not see a difference between component release and Kuadrant release in a high-level process. For each component there should be an RC build, and for each component QA should happen, exactly the way described here. The doc is also present for all components on the `kuadrant.io`, communication channels are also the same

So basically in my eyes, the process is the key section and it should apply to all component releases and the entire Kuadrant release at the same time.

[the remark this answers is not in the document]
Comment (guicassolato) [reply rfcs/0000-kuadrant-release-process.md:11]: @pehala, I think you may be seeing Kuadrant == MGC (or any other components that currently sits at the root of the tree of dependencies.)

In case my view wasn't clear... IMO, Kuadrant is a composition of things (Kuadrant Operator, Authorino, Limitador, MGC) and a way to use those things. Let's call this second part "other stuff" – docs, templates, maybe an installer, and, in general, other files we can/could provide in the "Kuadrant package" that are not part of the none of the individual components per se.

So:

```
K = KOp + A + L + MGC + stuff
```

A given version `K_v1` of Kuadrant can be, for example, a composition of Kuadrant Operator v0.3, Authorino v15, Limitador v0.8 and MGC v0.1. I.e.:

```
K_v1 = KOp_v0.3 + A_v15 + L_v0.8 + MGC_v0.1 + stuff
```

If we want to bump the version of, say, Limitador that is used in/by Kuadrant to v0.9 - not because Limitador v0.9 has just been released; maybe the latest Limitador is already v2.5 by the time this happens –, then the Kuadrant composition will change. IOW, Kuadrant will be a different thing than what it was at v1. Maybe we are talking about Kuadrant v2:

```
K_v2 = KOp_v0.3 + A_v15 + L_v0.9 + MGC_v0.1 + stuff
```

The only thing that changed was Limitador and, maybe, some of the other `stuff`, like new docs saying that now Kudrant v2 comes with Limitador v0.9.

Each component, as well as the suite itself, has (should have) its own set of artifacts, its own version number, its own change log and release notes, its own testing and release process. Kuadrant is no longer equivalent to the Kuadrant Operator. MGC may have taken its place as the new root, but we need to see (...and test, ...and version, ...and release) Kuadrant independently from it as well.

This is the way I see Kuadrant, at least. I may change my mind if most people agree that Kuadrant == MGC, in which case I would argue for changes beyond the defining our release process, starting by renaming those repos, MGC and Kuadrant Operator, and moving some code and responsibilities around. In the end, it would all contribute to defining what releasing Kuadrant means and how it should look like.

[the remark this answers is not in the document]
Comment (didierofrivia) [reply rfcs/0000-kuadrant-release-process.md:11]: I kind of like the idea you bring defining what would be the version of _Kuadrant_, quoting your formula:

> K = KOp + A + L + MGC + stuff

In that case, and if everyone/most people agree, I could add another section regarding the particular versioning and grouping that will define it.

**Comment by alexsnaps** [comment]
Added the "Final Comment Period" to this RFC, so please try to consolidate responses within the coming week... 
Other than major disagreement, we shall merge this after this period.

File: rfcs/0000-kuadrant-release-process.md [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+
+- Feature Name: `kuadrant-release-process`
+- Start Date: 2024-01-11
+- RFC PR: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/pull/0000)
+- Issue tracking: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Kuadrant is a set of components that their artifacts are built and delivered independently. This RFC aims to define every
```
Comment (thomasmaas) [inline rfcs/0000-kuadrant-release-process.md:11]:
```suggestion
Kuadrant is a set of components whose artifacts are built and delivered independently. This RFC aims to define every
```

File: rfcs/0000-kuadrant-release-process.md [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+A more detailed explanation of the versioning scheme can be found in the [Semantic Versioning](https://semver.org/) website.
+
+By releasing a new version of Kuadrant, we mean releasing the set of components with their corresponding semantic versioning,
+some of them maybe freshly released, or others still using versioning from the previous one, and being the version of the
+**Kuadrant Operator** the one that defines the version of the whole suite.
+
+The technical details of how to release each component are out of the scope of this RFC and could be found in the
+[Kuadrant components CI/CD](https://github.com/Kuadrant/architecture/pull/41) RFC.
+
+## Handover to QA
```
Comment (thomasmaas) [inline rfcs/0000-kuadrant-release-process.md:65]: I keep thinking that this should not be part of the release process upstream. We should make sure the test suite is always as up to date as possible and those tests that exist should pass. But that's it. Any gap that exists between features and QE coverage should be minimized by optimizing our development process, not by adding a post development QE process. By adding such a post development QE step we've already thrown in the towel.

Comment (pehala) [reply rfcs/0000-kuadrant-release-process.md:65]: But the reason for having a small QE part on the release is not about testsuite, I believe there are tasks which are simply unfeasible to run frequently (e.g. huge overhead) and hence are ideal for the release testing, for example, running on all supported configurations (single + multicluster).

Some tasks can only be done after we have a final bundle completed, like testing quickstart, verifying OLM deliverables, and verifying that all images are correct, which cannot be done beforehand.

Comment (thomasmaas) [reply rfcs/0000-kuadrant-release-process.md:65]: Ok, maybe the term "Handover to QE" is what triggered my response as it sounds heavy and slow to me. Also, do we "support configurations"? I'd say we "enable configurations". And if there are issues they can be fixed. I guess what I'm saying is that we should not drop the full weight of downstream on upstream. That doesn't mean we should ship sloppy software but it does mean we can make a different risk calculation.

Comment (pehala) [reply rfcs/0000-kuadrant-release-process.md:65]: I agree with both making a better name for the step and that we should not drop the full weight downstream on upstream. I believe the only thing that have not yet reached an agreement on is the risk calculation itself.

While it would be completely ok to ship without QE step at release at this stage. However, I believe the value of additional QE testing on release outweighs the delay it causes (e.g. the 2-3 days maximum that we commit ourselves to do). We are also definitely not treating it as a downstream release, we do have different requirements for upstream (only automation run, fixed time to test vs test everything etc).

I do not see the inclusion of the "QE" step in a release process as a downstream thing, but as a last sanity check that everything is ok and as a place to do testing on the all assembled pieces compared to individual components (Note: This argument does not hold for individual component releases, however, the still the last run of the automation on a final build is still a good sanity check). There is a fine balance of testing required that we need to find and what exactly should be tested for upstream but I do not think that eliminates the step, only reduces its length.

Regarding the `support configuration` vs `enable configurations`, enable is a better name for it for sure (I keep defaulting to downstream language because I have no idea how to call it otherwise), however, I think it is also an important part of the testing for a project where multicluster is a key feature, so testing it on dedicated multiple production-grade clusters makes sense, just not on every PR

Comment (thomasmaas) [reply rfcs/0000-kuadrant-release-process.md:65]: This is a helpful conversation and I think we're pretty close to agreeing. How about we call the step "QE sanity check"? We might also add that we aim for this step to take 0 extra time and to be fully automated at some moment in the future but that for now we agree on an expected duration of 1-2 days and a maximum duration of 3 days.

Comment (pehala) [reply rfcs/0000-kuadrant-release-process.md:65]: I am fully onboard with that idea! Thanks for suggesting it!

File: rfcs/0000-kuadrant-release-process.md [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+4. If for some reason it's needed to trigger the workflow manually, one can do it from the GitHub Actions tab in the
+[docs.kuadrant.io](https://github.com/Kuadrant/docs.kuadrant.io/actions/workflows/build.yaml) (`Actions > ci > Run Workflow`).
+
+## Communication
+
+Another important aspect of releasing a new version of the Kuadrant suite is the communication with the community and
+other teams within the organization. A few examples of the communication channels that need to be updated are:
+
+- Changelog generation
+- Release notes
```
Comment (eguzki) [inline rfcs/0000-kuadrant-release-process.md:173]: I like how Istio is doing this.

Release announcements

Screenshot 2024-02-16 at 14-59-30 Release Announcements


X.Y.x patch releases


Screenshot 2024-02-16 at 15-00-23 1 20 x Releases

New release notes

Screenshot 2024-02-16 at 15-00-52 Announcing Istio 1 20 3

File: rfcs/0000-kuadrant-release-process.md [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+
+Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.
+
+A more detailed explanation of the versioning scheme can be found in the [Semantic Versioning](https://semver.org/) website.
+
+By releasing a new version of Kuadrant, we mean releasing the set of components with their corresponding semantic versioning,
+some of them maybe freshly released, or others still using versioning from the previous one, and being the version of the
+**Kuadrant Operator** the one that defines the version of the whole suite.
+
+The technical details of how to release each component are out of the scope of this RFC and could be found in the
```
Comment (eguzki) [inline rfcs/0000-kuadrant-release-process.md:62]: This is the most interesting aspect for me. So I leave you (capable) guys deciding the other stuff

**Review by eguzki — COMMENTED** [review]
left some comments

File: rfcs/0000-kuadrant-release-process.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (pehala) [reply rfcs/0000-kuadrant-release-process.md:11]: I am all for the section defining Kuadrant and its related components. For the issue of the individual component release process, my suggestion would be to use this process for operator-based components that are released individually, meaning, this release process would be used for these components:

- Kuadrant as a whole
- Authorino + Authorino operator
- Limitador + Limitador Operator
- DNSOperator
- Multicluster Gateway Controller

These components are all operator-based and distributed the same way; thus, it makes sense for them to follow the same process.

@guicassolato @didierofrivia Is that ok with you? Does that even make sense?

[the remark this answers is not in the document]
Comment (didierofrivia) [reply rfcs/0000-kuadrant-release-process.md]: @pehala thanks for you comment! Let me elaborate and see if I understood correctly.... within a release of Kuadrant, one will encounter Operators released independently, which I reckon all of us agree (at least to a certain point) that given their similarities its easy to understand they will follow the same technical process of releasing... 

But given your last comment and trying to summarize the thread,  are we meaning to do...

```
Kuadrant vx.y.z = Kuadrant Operator vx.y.z + Authorino Operator vx.y.z + Limitador Operator vx.y.z + DNS Operator vx.y.z + MGC Controller vx.y.z
```
Where each dependency component will be released independently and grouped under a version of Kuadrant and then we could assign the Kuadrant version in the same fashion of the other components.

OR

Keeping the same way of independently releasing but matching the version of Kuadrant to its Kuadrant Operator:

```
Kuadrant vx.y.z = Kuadrant Operator vx.y.z =  Authorino Operator vx.y.z + Limitador Operator vx.y.z + DNS Operator vx.y.z + MGC Controller vx.y.z
```

Maybe I'm missing the point, but that is what we still haven't agreed on. 

@guicassolato @pehala how close are we to (dis)agree on this? Thanks in advance!

File: rfcs/0000-kuadrant-release-process.md [hunk 11]
```diff
@@ -0,0 +1,207 @@
...
+- [Authorino Operator](https://github.com/Kuadrant/authorino-operator): A Kubernetes Operator to manage Authorino instances.
+- [Limitador](https://github.com/Kuadrant/limitador): A generic rate-limiter written in Rust.
+- [Limitador Operator](https://github.com/Kuadrant/limitador-operator/): A Kubernetes Operator to manage Limitador deployments.
+- [Wasm Shim](https://github.com/Kuadrant/wasm-shim/): A Proxy-Wasm module written in Rust, acting as a shim between Envoy and Limitador.
+- [Multicluster Gateway Controller](https://github.com/Kuadrant/multicluster-gateway-controller): Provides multi-cluster
+connectivity and global load balancing.
+- [Kuadrant Operator](https://github.com/Kuadrant/kuadrant-operator/): The Operator to install and manage the lifecycle
+  of the Kuadrant components deployments.
+
+Each of them needs to be versioned independently, and the versioning scheme should follow [Semantic Versioning]. At
```
Comment (guicassolato) [inline rfcs/0000-kuadrant-release-process.md:46]:
```suggestion
Each of them needs to be versioned independently, and the versioning scheme should follow [Semantic Versioning](https://semver.org/). At
```

File: rfcs/0000-kuadrant-release-process.md [hunk 12]
```diff
@@ -0,0 +1,207 @@
...
+The technical details of how to release each component are out of the scope of this RFC and could be found in the
+[Kuadrant components CI/CD](https://github.com/Kuadrant/architecture/pull/41) RFC.
+
+## QA Sanity Check
+
+Probably the most important and currently missing step in the release process is the green flagging from the Quality Assurance
+(QA) team. The QA team is responsible for testing the different components of the Kuadrant suite, and they need to
+be aware of the new version of the suite that is going to be released, what are the changes that are included, bug fixes
+and new features in order they can plan their testing processes accordingly. This check is not meant to be a fully fledged
+assessment from the QA team as it would be with a downstream project when it's handover to them, it's aimed to not take
```
Comment (guicassolato) [inline rfcs/0000-kuadrant-release-process.md:71]:
```suggestion
assessment from the QA team when it's handover to them, it's aimed to not take
```

File: rfcs/0000-kuadrant-release-process.md [hunk 13]
```diff
@@ -0,0 +1,207 @@
...
+more than 1-2 days, and be expected to be fully automated in the future. This step will happen once the release candidate
```
Comment (guicassolato) [inline rfcs/0000-kuadrant-release-process.md:72]:
```suggestion
more than 1-2 days, and ideally expected to be fully automated. This step will happen once the release candidate
```

File: rfcs/0000-kuadrant-release-process.md [hunk 14]
```diff
@@ -0,0 +1,207 @@
...
+is ready, and it's been tested by the Engineering team. The QA team should be notified in advance of the release, and
+they should be able to assert the release candidate is ready before the actual release happens.
```
Comment (guicassolato) [inline rfcs/0000-kuadrant-release-process.md:74]:
```suggestion
```suggestion
has no PRs pending to be merged, and it has been tested by the Engineering team. The QA team should work closely to the engineering throughout the process, both teams aiming for zero handover time and continuous delivery mindset, so immediate testing can be triggered on release candidates once handed over. This process should happen without the need of formal communication between the teams or any overhead in general, but by keeping constant synergy between quality and product engineering instead.
``````

File: rfcs/0000-kuadrant-release-process.md [hunk 15]
```diff
@@ -0,0 +1,207 @@
...
+
+There is an ideal time to hand over to the QA team for testing, especially since we are using GitHub for orchestration,
+we could briefly define it in the following steps:
+
+1. Complete Development Work: The engineering team completes their work included in the milestone.
+2. Create Release Candidate: The engineering team creates Release Candidate builds and manifests for all components required
+for the release
+3. Notify QA Team: At this time, the engineering team can notify the QA team that the deliverables for the release candidate
```
Comment (guicassolato) [inline rfcs/0000-kuadrant-release-process.md:82]: I personally think this step should not exist.

File: rfcs/0000-kuadrant-release-process.md [hunk 16]
```diff
@@ -0,0 +1,207 @@
...
+are ready for testing. This notification can be done through the GitHub Kuadrant board, or simply a public message in the
+Kuadrant Slack channel.
+4. Flagging/Testing: The QA team do the actual assertion/testing of the release candidate, checking for any obvious bugs or issues.
+Then QA reports all the bugs as GitHub issues and communicates testing status back publicly on Slack and/or email.
+5. Iterate: Based on the feedback from the QA team, the development team makes any necessary adjustments and repeats the
```
Comment (guicassolato) [inline rfcs/0000-kuadrant-release-process.md:87]: To be consistent with the terminology, should we call it "engineering team", "development team", "product engineering team" (since QE is also "engineering")?

File: rfcs/0000-kuadrant-release-process.md [hunk 17]
```diff
@@ -0,0 +1,207 @@
...
+process until the release candidate is deemed ready for production.
+6. Publish Release: Once QA communicates that the testing has been successfully finished, the engineering team will publish
+the release both on Github and in the corresponding registries, updates documentation for the new release, and communicates
+it to all channels specified in Communication section.
+
+## Cadence
+
+Once the project is stable enough, and it's adoption increases, the community will be expecting a certain degree of
```
Comment (guicassolato) [inline rfcs/0000-kuadrant-release-process.md:95]:
```suggestion
Once the project is stable enough, and its adoption increases, the community will be expecting a certain degree of
```

File: rfcs/0000-kuadrant-release-process.md [hunk 18]
```diff
@@ -0,0 +1,207 @@
...
+it's been discussed in the past that it should be released every 3-4 weeks initially, including the latest released version
+of every component in the suite. There's another RFC that focuses on the actual frequency of each component, one could
+refer to the [Kuadrant Release Cadence RFC](https://github.com/pmccarthy/architecture/blob/release-cadence-proposal/rfcs/0007-project-release-cadence.md).
+
+There are a few reasons for this:
+
+- Delivering Unparalleled Value to Customers: Regular releases can provide customers with regular updates and improvements.
+  These updates can include new features and essential bug fixes, thus enhancing the overall value delivered to the customers.
+- Maximizing Deployment Efficiency: By releasing software at regular intervals, teams can align their activities with
+  available resources and environments, ensuring optimal utilization. This leads to increased efficiency in the deployment process.
```
Comment (guicassolato) [inline rfcs/0000-kuadrant-release-process.md:107]: Is this about cross product/project/dependency compatibility? What "available resources and environments" are we talking about? Or maybe it's not clear to me who the "teams" are in this case.

File: rfcs/0000-kuadrant-release-process.md [hunk 19]
```diff
@@ -0,0 +1,207 @@
...
+- Risk Management: Regular releases can help identify and fix issues early, reducing the risk of major failures that could
+  affect customers.
```
Comment (guicassolato) [inline rfcs/0000-kuadrant-release-process.md:109]:
```suggestion
  affect users.
```

File: rfcs/0000-kuadrant-release-process.md [hunk 20]
```diff
@@ -0,0 +1,207 @@
...
+|                                 | limitador-operator manifests                   | [OperatorHub.io](https://operatorhub.io/operator/limitador-operator)                   |
+| Wasm Shim                       | wasm-shim images                               | [Quay.io](https://quay.io/repository/kuadrant/wasm-shim)                               |
+| Multicluster Gateway Controller | multicluster-gateway-controller images         | [Quay.io](https://quay.io/repository/kuadrant/multicluster-gateway-controller)         |
+|                                 | multicluster-gateway-controller-bundle images  | [Quay.io](https://quay.io/repository/kuadrant/multicluster-gateway-controller-bundle)  |
+|                                 | multicluster-gateway-controller-catalog images | [Quay.io](https://quay.io/repository/kuadrant/multicluster-gateway-controller-catalog) |
+| Policy Controller               | policy-controller images                       | [Quay.io](https://quay.io/repository/kuadrant/policy-controller)                       |
```
Comment (guicassolato) [inline rfcs/0000-kuadrant-release-process.md:143]: Policy Controller was not mentioned as one of the components before.

File: rfcs/0000-kuadrant-release-process.md [hunk 21]
```diff
@@ -0,0 +1,207 @@
...
+
+The documentation for the Kuadrant suite is compiled and available on the [Kuadrant website](https://kuadrant.io/). One
+can find the source of the documentation within each component repository, in the `docs` directory. However, making this
+information available on the website is a manual process, and should be done by the maintainers of the project. The
+process of updating the documentation is simple and consists of the following steps:
+
+1. Update the documentation in the corresponding component repository.
+2. In [https://github.com/Kuadrant/docs.kuadrant.io/](https://github.com/Kuadrant/docs.kuadrant.io/), update the value of
+the `import_url` within the `multirepo` in the `plugins` section of the [mkdocs.yml](https://github.com/Kuadrant/docs.kuadrant.io/blob/main/mkdocs.yml)
+file, to point to the tag or branch of the component repository that contains the updated documentation.
```
Comment (guicassolato) [inline rfcs/0000-kuadrant-release-process.md:163]:
```suggestion
2. Follow the instruction in [https://github.com/Kuadrant/docs.kuadrant.io/](https://github.com/Kuadrant/docs.kuadrant.io/) to update the Docs pointers to the tag or branch of the component repository that contains the updated documentation.

```

File: rfcs/0000-kuadrant-release-process.md [hunk 22]
```diff
@@ -0,0 +1,207 @@
...
+The alternative to the proposal is to keep the current process, which is not ideal and leads to confusion and lack of
+transparency.
+
+# Prior art
+[prior-art]: #prior-art
+
+There's been an organically grown process for releasing new versions of the Kuadrant suite, which is not documented and
+it's been changing over time. However, there are some documentation for some of the components, worth mentioning:
+
+* [Authorino release process](https://docs.google.com/document/d/1tLveyv8Zwe0wKyfUTWOlEnFeMB5aVGqIVDUjVYWax0U)
```
Comment (guicassolato) [inline rfcs/0000-kuadrant-release-process.md:194]: I'm afraid this document is not accessible outside of Red Hat bu default. Shall we follow Limitador and move it into the Authorino repo?

File: rfcs/0000-kuadrant-release-process.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (pehala) [reply rfcs/0000-kuadrant-release-process.md]: When it comes to this, I have no preference, so whatever works for you

File: rfcs/0000-kuadrant-release-process.md (hunk 18 above)
Comment (didierofrivia) [reply rfcs/0000-kuadrant-release-process.md:109]: In this case teams would refer to (as of today) Engineering and QA, and in the future probably include Product.  Of course this makes more sense when talking about Downstream, but I think it could apply to Upstream too.

File: rfcs/0000-kuadrant-release-process.md (hunk 20 above)
Comment (didierofrivia) [reply rfcs/0000-kuadrant-release-process.md]: Now removed, since the Kuadrand Operator will manage the DNS and TLS policies. Also added the DNS Operator

File: rfcs/0000-kuadrant-release-process.md (hunk 15 above)
Comment (didierofrivia) [reply rfcs/0000-kuadrant-release-process.md:84]: Do you reckon it could be implicitly in the following step "Flagging/Testing" ?

File: rfcs/0000-kuadrant-release-process.md (hunk 18 above)
Comment (guicassolato) [reply rfcs/0000-kuadrant-release-process.md:109]: I think I know what mean. I first read this from a different angle, of users who have Kuadrant in their CI pipelines.

If it's about Kuadrant teams, "deployment" seems a little out of place IMO, but maybe it's me.

Perhaps "Resource Optimization"? As in teams working together and in sync can plan their resources better (e.g. avoid idle servers, etc?)

Could it be overlapping with _Synchronization_ a little?

File: rfcs/0000-kuadrant-release-process.md [hunk 23]
```diff
@@ -0,0 +1,211 @@
...
+
+The documentation for the Kuadrant suite is compiled and available on the [Kuadrant website](https://kuadrant.io/). One
+can find the source of the documentation within each component repository, in the `docs` directory. However, making this
+information available on the website is a manual process, and should be done by the maintainers of the project. The
+process of updating the documentation is simple and consists of the following steps:
+
+1. Update the documentation in the corresponding component repository.
+2. Follow the instruction in [https://github.com/Kuadrant/docs.kuadrant.io/](https://github.com/Kuadrant/docs.kuadrant.io/)
+to update the Docs pointers to the tag or branch of the component repository that contains the updated documentation.
+file, to point to the tag or branch of the component repository that contains the updated documentation.
```
Comment (guicassolato) [inline rfcs/0000-kuadrant-release-process.md:166]:
```suggestion
```

**Review by guicassolato — APPROVED** [review]
Nice job, @didierofrivia!

File: rfcs/0000-kuadrant-release-process.md [hunk 24]
```diff
@@ -0,0 +1,211 @@
+# Kuadrant Release Process
+
+- Feature Name: `kuadrant-release-process`
+- Start Date: 2024-01-11
+- RFC PR: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/pull/0000)
+- Issue tracking: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/issues/0000)
```
Comment (guicassolato) [inline rfcs/0000-kuadrant-release-process.md:6]: Please don't forget to update these 😉

File: rfcs/0000-kuadrant-release-process.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (guicassolato) [reply rfcs/0000-kuadrant-release-process.md]:
```
Kuadrant vX.Y.Z = Kuadrant Operator vX.Y.Z =  Authorino Operator vA.B.C + Limitador Operator vD.E.F + DNS Operator vG.H.I + MGC Controller vJ.K.L
```

**Review by pehala — APPROVED** [review]

**Review by guicassolato — APPROVED** [review]
